### PR TITLE
Red Brinstar runway migration

### PR DESCRIPTION
**Pull request by blkerby**

### CODE REVIEW

File: region/brinstar/red.json [hunk 1]
```diff
@@ -1199,6 +1224,26 @@
...
+                  "notable": false,
+                  "entranceCondition": {
+                    "comeInShinecharging": {
+                      "length": 6,
+                      "openEnd": 1
+                    }
+                  },
+                  "requires": [
+                    {"shinespark": {
+                      "frames": 150,
```
Comment (kjbranch) [inline region/brinstar/red.json:1238]: 146 frames through the door, 60 excess frames
at the end of the runway is 142 frames, with 60 excess

**Review by kjbranch — COMMENTED** [review]
Here's part of my review, I'll look through more later.

File: region/brinstar/red.json [hunk 2]
```diff
@@ -2453,6 +2493,43 @@
...
+                    "Morph",
+                    {"obstaclesCleared": ["A"]},
+                    {"shinespark": {"frames": 22}}
+                  ],
+                  "exitCondition": {
+                    "leaveWithSpark": {}
+                  },
+                  "note": "Slow rolling or getting a fast mockball work equally well.",
+                  "devNote": [
+                    "One tile longer runway could be used if the door to the right of the gate were known to be unlocked"
```
Comment (kjbranch) [inline region/brinstar/red.json:2528]:
```suggestion
                    "FIXME: One tile longer runway could be used if the door to the right of the gate were known to be unlocked"
```

File: region/brinstar/red.json [hunk 3]
```diff
@@ -2495,13 +2572,47 @@
...
+                  "exitCondition": {
+                    "leaveWithRunway": {
+                      "length": 9,
+                      "openEnd": 0
+                    }
+                  },
+                  "note": [
+                    "Runway not usable if the shot blocks are broken.",
+                    "Those blocks are not an obstacle right now, but there's no need to destroy them except when travelling between nodes 2 and 3."
+                  ]
```
Comment (kjbranch) [inline region/brinstar/red.json:2604]:
```suggestion
                  "note": "Runway not usable if the shot blocks are broken."
```

File: region/brinstar/red.json [hunk 4]
```diff
@@ -3070,6 +3217,33 @@
...
+                      "openEnd": 0
+                    }
+                  },
+                  "note": [
+                    "This runway requires the PB blocks to be intact, but the sidehoppers to be dead.",
+                    "This is relevant because it can limit the player's options for killing the Sidehoppers."
+                  ],
+                  "devNote": [
+                    "Killing the Sidehoppers requires going to 3 and back.",
+                    "FIXME: A shorter runway could be added assuming a PB was laid as far left as possible."
```
Comment (kjbranch) [inline region/brinstar/red.json:3242]: placing a PB against the left wall kills the Sidehoppers, or if some survive, they can safely be killed with pea-shooter. The remaining runway is 12 tiles + 1 open end

**Review by osse101 — DISMISSED** [review]

**Comment by blkerby** [comment]
Looks good!

**Pull request merged**